[BUG] Scaladoc link warnings in ParquetCachedBatchSerializer and ExplainPlan

**Describe the bug**
Seeing the following scaladoc warnings emitted in a 21.12 build:
```
/home/ubuntu/spark-rapids/sql-plugin/src/main/311+-all/scala/com/nvidia/spark/ParquetCachedBatchSerializer.scala:105: warning: Could not find any member to link for "SimpleMetricsCachedBatchSerializer".
  /**
  ^
/home/ubuntu/spark-rapids/sql-plugin/src/main/311+-all/scala/com/nvidia/spark/ParquetCachedBatchSerializer.scala:105: warning: Could not find any member to link for "SimpleMetricsCachedBatch".
  /**
  ^
/home/ubuntu/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ExplainPlan.scala:29: warning: Could not find any member to link for "IllegalArgumentException".
  /**
  ^
/home/ubuntu/spark-rapids/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ExplainPlan.scala:29: warning: Could not find any member to link for "IllegalStateException".
  /**
  ^
four warnings found
```

**Steps/Code to reproduce bug**
`mvn verify -Dbuildver=311`

**Expected behavior**
No scaladoc warnings during the build

--- a/pom.xml
+++ b/pom.xml
@@ -734,6 +734,7 @@
         <buildver>301</buildver>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
+        <java.major.version>8</java.major.version>
         <spark.version>${spark301.version}</spark.version>
         <spark.test.version>${spark.version}</spark.test.version>
         <spark.version.classifier>spark${buildver}</spark.version.classifier>
@@ -1031,7 +1032,9 @@
                             </goals>
                             <configuration>
                                 <args>
+                                    <arg>-doc-external-doc:${java.home}/lib/rt.jar#https://docs.oracle.com/javase/${java.major.version}/docs/api/index.html</arg>
                                     <arg>-doc-external-doc:${settings.localRepository}/${scala.local-lib.path}#https://scala-lang.org/api/${scala.version}/</arg>
+                                    <arg>-doc-external-doc:${settings.localRepository}/org/apache/spark/spark-sql_${scala.binary.version}/${spark.version}/spark-sql_${scala.binary.version}-${spark.version}.jar#https://spark.apache.org/docs/${spark.version}/api/scala/index.html</arg>
                                 </args>
                             </configuration>
                         </execution>

--- a/sql-plugin/src/main/311+-all/scala/com/nvidia/spark/ParquetCachedBatchSerializer.scala
+++ b/sql-plugin/src/main/311+-all/scala/com/nvidia/spark/ParquetCachedBatchSerializer.scala
@@ -132,9 +132,10 @@ class ParquetCachedBatchSerializer extends GpuCachedBatchSerializer {
 
   /**
    * Builds a function that can be used to filter batches prior to being decompressed.
-   * In most cases extending [[SimpleMetricsCachedBatchSerializer]] will provide the filter logic
-   * necessary. You will need to provide metrics for this to work. [[SimpleMetricsCachedBatch]]
-   * provides the APIs to hold those metrics and explains the metrics used, really just min and max.
+   * In most cases extending [[org.apache.spark.sql.columnar.SimpleMetricsCachedBatchSerializer]]
+   * will provide the filter logic necessary. You will need to provide metrics for this to work.
+   * [[org.apache.spark.sql.columnar.SimpleMetricsCachedBatch]] provides the APIs to hold those
+   * metrics and explains the metrics used, really just min and max.
    * Note that this is intended to skip batches that are not needed, and the actual filtering of
    * individual rows is handled later.
    * @param predicates the set of expressions to use for filtering.

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ExplainPlan.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/ExplainPlan.scala
@@ -53,9 +53,9 @@ object ExplainPlan {
    * @param explain If ALL returns all the explain data, otherwise just returns what does not
    *                work on the GPU. Default is ALL.
    * @return String containing the explained plan.
-   * @throws IllegalArgumentException if an argument is invalid or it is unable to determine the
-   *         Spark version
-   * @throws IllegalStateException if the plugin gets into an invalid state while trying
+   * @throws java.lang.IllegalArgumentException if an argument is invalid or it is unable to
+   *         determine the Spark version
+   * @throws java.lang.IllegalStateException if the plugin gets into an invalid state while trying
    *         to process the plan or there is an unexepected exception.
    */
   @throws[IllegalArgumentException]

